tests/framework/installer/install_helper.go

```diff
@@ -216,75 +249,93 @@ func (h *InstallHelper) InstallRookOnK8sWithHostPathAndDevices(namespace, storeT
+		assert.NoError(h.T(), err, "cannot delete TPRs. %+v", err)
+	}
 
 	if helmInstalled {
 		err = h.helmHelper.DeleteLocalRookHelmChart(helmDeployName)
-		assert.NoError(h.T(), err, "cannot uninstall Rook helm chart err -> %v", err)
 	} else {
-		rookOperator := h.installData.GetRookOperator(SystemNamespace(namespace))
+		rookOperator := h.installData.GetRookOperator(systemNamespace)
 		_, err = h.k8shelper.KubectlWithStdin(rookOperator, deleteArgs...)
-		assert.NoError(h.T(), err, "cannot uninstall rook-operator err -> %v", err)
 	}
+	// The rook system namespace is deleted with the delete command, which may remove the operator deployment and return a "not found" error when the deployment is deleted
+	assert.True(h.T(), err == nil || errors.IsNotFound(err), fmt.Sprintf("cannot uninstall rook-operator. %v", err))
 
 	h.k8shelper.Clientset.RbacV1beta1().ClusterRoles().Delete("rook-agent", nil)
 	h.k8shelper.Clientset.RbacV1beta1().ClusterRoleBindings().Delete("rook-agent", nil)
+	h.k8shelper.Clientset.RbacV1beta1().ClusterRoleBindings().Delete("anon-user-access", nil)
+	logger.Infof("done removing the operator from namespace %s", systemNamespace)
+}
 
-	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "cluster", namespace})
-	assert.NoError(h.T(), err, "cannot remove cluster %s in namespace %s  err -> %v", namespace, namespace, err)
-
-	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "serviceaccount", "rook-api"})
-	assert.NoError(h.T(), err, "cannot remove serviceaccount rook-api in namespace %s  err -> %v", namespace, err)
-
-	_, err = k8sHelp.DeleteResource([]string{"-n", namespace, "serviceaccount", "rook-ceph-osd"})
-	assert.NoError(h.T(), err, "cannot remove serviceaccount rook-ceph-osd in namespace %s  err -> %v", namespace, err)
-
-	if h.k8shelper.VersionAtLeast("v1.6.0") {
-		err = h.k8shelper.DeleteRoleAndBindings("rook-api", namespace)
-		assert.NoError(h.T(), err, "rook-api cluster role and binding cannot be deleted: %+v", err)
-
-		err = k8sHelp.DeleteRoleAndBindings("rook-ceph-osd", namespace)
-		assert.NoError(h.T(), err, "rook-ceph-osd cluster role and binding cannot be deleted: %+v", err)
+func (h *InstallHelper) waitForCustomResourceDeletion(namespace string) error {
+	if !h.k8shelper.VersionAtLeast("v1.7.0") {
+		// there is no finalizer for TPRs so we can return immediately
+		return nil
 	}
 
-	if h.k8shelper.VersionAtLeast("v1.7.0") {
-		_, err = k8sHelp.DeleteResource([]string{"crd", "clusters.rook.io", "pools.rook.io", "objectstores.rook.io", "filesystems.rook.io"})
-		assert.NoError(h.T(), err, "cannot delete CRDs err -> %+v", err)
-	} else {
-		_, err = k8sHelp.DeleteResource([]string{"thirdpartyresources", "cluster.rook.io", "pool.rook.io", "objectstore.rook.io", "filesystem.rook.io"})
-		assert.NoError(h.T(), err, "cannot delete TPRs err -> %+v", err)
+	// wait for the operator to finalize and delete the cluster CRD
+	for i := 0; i < 10; i++ {
```

Comment:
we should probably use i < utils.RetryLoop instead of  i<10 so all retires are consistent.

Reply:
i think the RetryLoop const works well when it is in a helper package where the implementation doesn't likely know how long an operation might take in a specific scenario. But when the timeout is for a specific scenario, I would recommend we use consts like this that are more tuned for the scenario.